rpc/service.go

```diff
@@ -18,6 +21,13 @@ type RPCHandler interface {
 	AddOrder(order *zeroex.SignedOrder) error
 	// AddPeer is called when the client sends an AddPeer request.
 	AddPeer(peerInfo peerstore.PeerInfo) error
+	// Orders is called when a client sends a Subscribe to orderStream request
+	Orders(ctx context.Context) (*rpc.Subscription, error)
+}
+
+// Orders calls rpcHandler.Orders and returns the rpc subscription.
```

Comment:
The `go-ethereum/rpc` package has some very specific requirements about the type signature and name of methods which reduces the flexibility of our API a lot. The purpose of __rpc/service.go__ is to smooth out this API and expose names and signatures that make more sense in the context of the entire application. `rpcService` implements the API required by the `go-ethereum/rpc` package, but `RPCHandler` doesn't have to and is not necessarily bound by the same API restrictions.

At the very least, I think we should rename `RPCHandler.Orders` to something like `RPCHandler.SubscribeToOrders`. The name `Orders` is too generic and doesn't make sense in the context of the broader application.

We should also consider simplifying the API. For example, if we update the `core` package so that it is not RPC-aware, we could also modify `RPCHandler` to be more aligned with the `core` package API to reduce the required glue code (e.g. `RPCHandler` could have a method that accepts a `chan []*zeroex.OrderInfo` and `rpcService` can wrap this up into an `rpc.Subscription`). I'm okay with punting on changing the API until after we make progress on browser support, because that will affect what the common API in the `core` package looks like.